### ISSUE TITLE
Questions about VCBM Reproducibility, Dataset Alignment, and MViTv2 Evaluation

### ISSUE THREAD

**Issue by Yuyan-Bin** [Mukil07/Explainable-Driver-Intention-Prediction#3]
Dear Authors,

Thank you for releasing the DIP-Explainability (VCBM) code, weights, and DAAD-X dataset. I am currently working on reproducing the results, but I have encountered a few questions regarding the provided checkpoints, data alignment, and evaluation commands:

**1. Correspondence of HuggingFace weights and low metrics**

 I am using the two .pth files from HuggingFace. With the full proposed command, the model runs successfully (Total params: ~24.78M, 35/35 batches), but the metrics are quite low:

- accuracy and F1(GAZE): 0.0714/ 0.0277
- accuracy and F1(EGO): 0.0179 / 0.0304
- EGO Macro F1 and Micro F1: 0.0164 / 0.0323
- Main accuracy and F1: 0.51071/ 0.4818 (around 50%).

Do these provided weights strictly correspond to the paper's reported results? If so, could there be any specific data preprocessing or seed settings causing the Gaze/EGO metrics to collapse near zero while the main task stays at ~55%?

<img width="1904" height="471" alt="Image" src="https://github.com/user-attachments/assets/fd67a863-3e9c-4bc6-b5df-0a027ae3b52f" />

**2. Missing 28 videos in DAAD-X alignment**

I noticed that front_view_common is missing 28 corresponding scenario videos compared to ariagaze_view_common. The filenames of the missing videos are listed below.

1. "a1f968c2-b34a-45a1-b3a8-3ba1bd3826e6.mp4",
2. "51e01869-2072-4b2f-9812-eee04796ebc1.mp4",
3. "3e26e45f-6049-48a9-92ff-b7405eae33de.mp4",
4. "108fc435-8508-4c08-9f06-cd0cf388c4f2.mp4",
5. "d3988275-0894-4b6b-8b31-4af7c0500873.mp4",
6. "8085b347-a8fc-453a-a77e-80ecace3b40c.mp4",
7. "16081d27-b46b-49a1-81d9-43bbb9ee2a20.mp4",
8. "7e8f69d3-5596-4124-923b-e8883150845f.mp4",
9. "074a8863-aac6-4027-9279-7e82a4e16fde.mp4",
10. "453a4fde-f4ba-4071-8c5c-24a0b9844fa7.mp4",
11. "a6228ac3-8b54-45e8-8060-8a20eb1d6a77.mp4",
12. "3261b14a-216f-4e39-9d9d-623630dbb242.mp4",
13. "1898f598-9d73-47ff-a629-c4ea6ffcc3cd.mp4",
14. "236c2758-9f3b-48bb-9a99-c2a48b0b6779.mp4",
15. "ebdd1080-2e2c-4e90-b161-9db7e761d40c.mp4",
16. "fd123097-4b65-4c26-8af5-0420df4fb952.mp4",
17. "e2fd33ba-aad3-4de8-a66f-b37d94c3024c.mp4",
18. "9093c116-a7ae-4a32-953f-54e31f9e413f.mp4",
19. "3196a213-404b-435f-943f-0beaf90ed239.mp4",
20. "48722287-b38a-41de-8a20-c417b028a25d.mp4",
21. "dbd10536-8920-434f-842c-20e34dacc763.mp4",
22. "9110f33c-3337-49f5-aace-ed761c9acab1.mp4",
23. "c10c4cd1-57b7-4cb3-ba51-8efb0bfd5672.mp4",
24. "fa4e458e-4f59-4db9-9225-9a3edddc0f78.mp4",
25. "545fd3be-4db8-478d-ab57-ecfb9a9bafdf.mp4",
26. "4ff44479-4677-4571-8ab7-2cc2591f4f33.mp4",
27. "366e05e2-308c-4d44-a8e7-5ab1cd942d60.mp4",
28. "14a0ccf2-287b-4934-9fa9-44574f8cb28b.mp4",

**3. Missing evaluation command for MViTv2**

The README.md does not mention the evaluation command for the MViTv2 corresponding weights. Could you please provide the exact eval.py arguments to properly evaluate MViTv2?

Thank you for your time and for sharing this great work.

**Comment by Mukil07** [comment]
Hi @Yuyan-Bin, apologies for the late reply. The weights uploaded in the huggingface was without the composite similarity finetuned block, which was the reason for not performing well. Please find the new checkpoint [here](https://huggingface.co/Skyrmion/VCBM/blob/main/i3d/i3d_dipx.pth) 

I have also updated the eval scripts, to make sure to use the composite similarity block, fix the clustering to 5, and, most importantly, use a batch size of 8. We have averaged the scores across different seeds. 

You can use this for eval,

```
MODEL=i3d_proposed
DATASET=dipx
TECH=i3d_proposed

python eval.py \
  --model $MODEL \
  --weights $WEIGHT \
  --batch 8 \
  --num_classes 7 \
  --dataset $DATASET \
  --technique $TECH  \
  --n_attributes 17  \
  --multitask_classes 15 \
  --clusters 5 \
  --csv DATA/val.csv \
  -ego_cbm -multitask -bottleneck
```

With the new checkpoint, we can get these results: 

<img width="1403" height="323" alt="Image" src="https://github.com/user-attachments/assets/f8a511db-c9fb-4c90-bb53-3515146a950d" />

<img width="1403" height="323" alt="Image" src="https://github.com/user-attachments/assets/b34c660a-5f6a-4bce-aa73-a463d87c5886" />

Regarding the dataset, please use the 285 common videos for the evaluation. Right now, we are only keeping the ego explanations as the bottleneck and we dont tune the gaze head, which is why the performance is bad for gaze. We left gaze explanations for future work. Please let me know if there are any issues.

**Comment by Mukil07** [comment]
For the training recipe, all experiments were conducted on a single RTX A6000 48GB GPU with a fixed batch size of 8. We used the AdamW optimizer with a weight decay of 0.05, a base learning rate of 5 × 10^-5, and a dropout rate of 0.45. We finetune the attention layers of the backbone, the final MLP classifiers, and our bottleneck blocks. VideoMAE and MViTv2 are trained for 200 epochs, while I3D is trained for 100 epochs, the plots above show the validation score at training epochs. Furthermore, we apply a cosine annealing learning rate scheduler with warm restarts, starting at 50 epochs with a reset multiplier of 2 for the VideoMAE and MViTv2 models.
